app-config.yaml

```diff
@@ -67,6 +67,12 @@ proxy:
     headers:
       Cookie: ${ARGOCD_AUTH_TOKEN}
 
+  '/prometheus/api':
+    # url to the api of your hosted prometheus instance
+    target: http://localhost:9090/api/v1/
```

Comment:
another question - can this be somehow overriden through annotations instead in the app-config? internally we have hundreds of prometheus servers deployed, through kubernetes namespaces. that would mean service A and service B send data to different servers (and the metrics are all aggregated by Thanos). assuming Thanos and Prometheus don't have a matching REST API Spec, as an integrator, I would like to be able to specify which path to query for each of my entities (or group of entities, somehow).

how could we model this?

Reply:
Yes, might need some pointers with the implementation details, but I believe I could, yes.